sources/api/corndog/src/main.rs

```diff
@@ -104,23 +109,83 @@ where
     path
 }
 
-/// Applies the requested sysctls to the system.  The keys are used to generate the appropriate
-/// path, and the value its contents.
-fn set_sysctls<K>(sysctls: HashMap<K, String>)
+/// Validates sysctl key/value pairs for systemd-sysctl compatibility
+fn validate_sysctl_entry<K>(key: K, value: &str) -> Result<()>
+where
+    K: AsRef<str>,
+{
+    let key = key.as_ref();
+
+    // Basic validation of key format
+    if !key.chars().all(|c| c.is_ascii_alphanumeric() || c == '.' || c == '_' || c == '-' || c == '*') {
+        return error::InvalidSysctlKeySnafu { key: key.to_string() }.fail();
+    }
+
+    // Ensure value doesn't contain newlines which could break the config file format
+    if value.contains('\n') {
+        return error::InvalidSysctlValueSnafu {
+            key: key.to_string(),
+            value: value.to_string()
+        }.fail();
+    }
+
+    Ok(())
+}
+
+/// Restarts systemd-sysctl service
+fn restart_sysctl() -> Result<()> {
+    let status = Command::new("systemctl")
+        .args(["restart", "systemd-sysctl"])
+        .status()
+        .context(error::SystemctlSnafu)?;
+
+    if !status.success() {
+        return error::SystemctlFailedSnafu.fail();
+    }
+
+    Ok(())
+}
```

Comment:
Restarting a service can lead to unexpected behavior if units depend on that service, or if for whatever reason it hasn't started yet or failed to start successfully. This is also using the blocking form which can lead to trouble in early boot.

It'd be better to avoid all of those issues and just invoke `systemd-sysctl` directly. I read your other comment and understand that we can't pass just the one config file, which makes sense, but the unit just runs `/usr/lib/systemd/systemd-sysctl` with no arguments which we can do here also.